tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -597,25 +627,54 @@ shared_ptr<DuckDBPyConnection> DuckDBPyConnection::Connect(const string &databas
 		}
 		config.SetOption(*config_property, Value(val));
 	}
-	res->database = make_unique<DuckDB>(database, &config);
-	res->connection = make_unique<Connection>(*res->database);
+
 	if (config.enable_external_access) {
-		res->database->instance->config.replacement_scans.emplace_back(ScanReplacement);
-	}
+		config.replacement_scans.emplace_back(ScanReplacement);
+	}
+	if (abs_path == ":memory:") {
+		new_instance = true;
+		return make_shared<DuckDB>(database, &config);
```

Comment:
Perhaps a bit orthogonal, but I wonder if it would be useful to allow connecting to in-memory databases through appending a unique number, e.g. `:memory:1`, `:memory:2` (where `:memory:` would instead always create a new instance).